### PR TITLE
Emit new package-related properties in generated C# project file.

### PR DESCRIPTION
**Pull request by mpiroc**
* Update JsonModel.Assembly to include new language-agnostic package-related properties.
* Update JsonModel.DotNetTarget to include new dotnet-specific package-related properties.
* Convert new properties to MSBuild properties and include them in generated `.csproj` files.

### CODE REVIEW

File: packages/jsii-dotnet-jsonmodel/src/Amazon.JSII.JsonModel.UnitTests/Spec/AssemblyTests.cs [hunk 1]
```diff
@@ -21,27 +21,85 @@ public void ShouldSerializeAllMembers()
...
                     version: "myVersion",
-                    types: new Dictionary<string, JsonModel.Spec.Type>(),
-                    dependencies : new Dictionary<string, PackageVersion>(),
+                    license: "Apache-2.0",
+                    targets: new AssemblyTargets(new AssemblyTargets.DotNetTarget(
+                        @namespace: "Dot.Net.Namespace",
+                        packageId: "Dot.Net.PackageId",
+                        signAssembly: true,
+                        assemblyOriginatorKey: "key.snk",
+                        tags: new [] { "tag1", "tag2" },
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-dotnet-jsonmodel/src/Amazon.JSII.JsonModel.UnitTests/Spec/AssemblyTests.cs]: https://github.com/awslabs/jsii/issues/170

File: packages/jsii-pacmak/test/expected.jsii-calc-base/dotnet/Amazon.JSII.Tests.Calculator.Base/Amazon.JSII.Tests.Calculator.Base.csproj [hunk 2]
```diff
@@ -2,9 +2,13 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <Authors></Authors>
-    <Company></Company>
     <PackageVersion>0.6.4</PackageVersion>
+    <PackageId>Amazon.JSII.Tests.Calculator.Base</PackageId>
+    <Description>@scope/jsii-calc-base</Description>
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-pacmak/test/expected.jsii-calc-base/dotnet/Amazon.JSII.Tests.Calculator.Base/Amazon.JSII.Tests.Calculator.Base.csproj]: `package.json` does not contain a `description`, so the jsii compiler falls back on `name`:

```typescript
// default "description" to "name" and "homepage" to repo url
if (!pkg.description) {
    pkg.description = pkg.name;
}
```

I'll add `description` properties to the `jsii-calc` packages.

File: packages/jsii-calc-base-of-base/package.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
         }
       },
       "dotnet": {
-        "namespace": "Amazon.JSII.Tests.Calculator.BaseOfBase"
+        "namespace": "Amazon.JSII.Tests.Calculator.BaseOfBase",
+        "packageId": "Amazon.JSII.Tests.Calculator.BaseOfBase"
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-calc-base-of-base/package.json]: ✔️

File: packages/jsii-pacmak/test/expected.jsii-calc-base/dotnet/Amazon.JSII.Tests.Calculator.Base/Amazon.JSII.Tests.Calculator.Base.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-pacmak/test/expected.jsii-calc-base/dotnet/Amazon.JSII.Tests.Calculator.Base/Amazon.JSII.Tests.Calculator.Base.csproj]: ✔️

File: packages/jsii-dotnet-jsonmodel/src/Amazon.JSII.JsonModel.UnitTests/Spec/AssemblyTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-dotnet-jsonmodel/src/Amazon.JSII.JsonModel.UnitTests/Spec/AssemblyTests.cs]: ✔️ (removed tags for now, refer to #170)

**Comment by eladb** [comment]
Please make sure to update README with the new dotnet target configuration schema

File: packages/jsii-ruby-runtime/package.json [hunk 4]
```diff
@@ -1,35 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-ruby-runtime/package.json]: ✔️

File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/composition/CompositionStringStyle.cs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 using Amazon.JSII.Runtime.Deputy;
 
-namespace Amazon.JSII.Tests.Calculator.composition
+namespace Amazon.JSII.Tests.CalculatorNamespace.composition
```
[the remark this answers is not in the document]
Comment (mpiroc) [reply packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.CalculatorPackageId/Amazon/JSII/Tests/CalculatorNamespace/composition/CompositionStringStyle.cs]: https://github.com/awslabs/jsii/issues/174

File: README.md [hunk 6]
```diff
@@ -86,7 +86,8 @@ Edit your `package.json`:
         }
       },
       "dotnet": {
-        "namespace": "Acme.Hello"
+        "namespace": "Acme.Hello",
+        "packageId": "Acme.Hello"
```
Comment (eladb) [inline README.md]: Set a different packageId here to make it clear where this value is going

File: README.md [hunk 7]
```diff
@@ -265,7 +267,15 @@ The following targets are currently supported:
 ```json
```
Comment (eladb) [inline README.md]: Change to `js` so that comments won't appear as errors

File: README.md [hunk 8]
```diff
@@ -265,7 +267,15 @@ The following targets are currently supported:
...
 {
   "dotnet": {
-    "namespace": "Acme.Hello"
+    /* required */
+    "namespace": "Acme.Hello",
+    "packageId": "Acme.Hello",
+
+    /* optional */
+    "title": "ACME Hello",
```
Comment (eladb) [inline README.md]: Why do we have this? Isn't this the same as description? If it's not provided, what is the default?

File: README.md [hunk 9]
```diff
@@ -265,7 +267,15 @@ The following targets are currently supported:
...
+    "iconUrl": "path/to/icon.svg",
```
Comment (eladb) [inline README.md]: Add comments with defaults to all fields

File: README.md (hunk 7 above)
Comment (mpiroc) [reply README.md]: ✔️

File: README.md (hunk 8 above)
Comment (mpiroc) [reply README.md]: It is not--`title` is a short human-readable name, while description is more detailed. If a NuGet package has no title, it defaults to the package's `packageId`. Updated comments to reflect this.

✔️

File: README.md (hunk 9 above)
Comment (mpiroc) [reply README.md]: ✔️

File: README.md (hunk 6 above)
Comment (mpiroc) [reply README.md]: ✔️

**Pull request merged**